### PR TITLE
[Feat] Synthetix Debt Pool - Optimism Bridge

### PR DESCRIPTION
**Pull request by justinkaseman**
## Description
Updates the existing debt pool oracle code to enable the transfer of sUSD for deposit and withdrawal through Optimism Bridge.

The issued synth formula has changed to:
```
sum(DebtCache.currentDebt() + SynthetixBridge.synthTransferReceived() - SynthetixBridge.synthTransferSent()) for all chains
```

*Note that the contract calls should be atomic or locked to a specific block on-chain.

## Changes

- Add ABI & name helper for SynthetixBridge
- Pin contract calls to the latest block of each chain at execution time
- Modify issued synths formula to use SynthetixBridge data
- Utilize proxy contracts for Address Resolver contracts

## Steps to Test

`yarn test packages/sources/synthetix-debt-pool/test/integration/adapter.test.ts`

## Quality Assurance

- [x] Ran `yarn changeset` if adapter source code was changed
- [x] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `<ADAPTER_PACKAGE>/schemas/env.json` and `<ADAPTER_PACKAGE>/README.md`
- [x] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `infra-k8s` configuration file.
- [x] The branch naming follows git flow (`feature/x`, `chore/x`, `release/x`, `hotfix/x`, `fix/x`) or is created from Clubhouse/Shortcut
- [x] This is related to a maximum of one Clubhouse/Shortcut story or GitHub issue
- [x] Types are safe (avoid TypeScript/TSLint features like any and disable, instead use more specific types)

### CODE REVIEW

File: packages/sources/synthetix-debt-pool/src/endpoint/debt.ts [hunk 1]
```diff
@@ -34,8 +49,22 @@ const getTotalDebtIssued = async (
...
+          getSynthetixBridgeName(network, jobRunID),
+        )
+        const synthetixBridge = new ethers.Contract(
+          synthetixBridgeAddress,
+          SYNTHETIX_BRIDGE_ABI,
+          networkProvider,
+        )
+        const synthTransferReceived = await synthetixBridge.synthTransferReceived({ blockNumber })
+        const synthTransferSent = await synthetixBridge.synthTransferSent({ blockNumber })
+        const issuedSynths = debtIssued.add(synthTransferReceived.sub(synthTransferSent))
```
Comment (dbeal-eth) [inline packages/sources/synthetix-debt-pool/src/endpoint/debt.ts:66]: hi,

I know you followed our instructions for the equation for the function, but it turns out that this equation was inverse of what it should be. 

```
const issuedSynths = debtIssued.add(synthTransferSent.sub(synthTransferReceived))
```

we will update the text of the SIP

apologies for the confusion

File: packages/sources/synthetix-debt-pool/test/integration/__snapshots__/adapter.test.ts.snap [hunk 2]
```diff
@@ -3,43 +3,43 @@
 exports[`synthetix-debt-pool debt successfully fetches the current debt size of the synthetix debt cache across "mainnet" and "mainnet-ovm" if chainSources is missing 1`] = `
 Object {
   "data": Object {
-    "result": "313273658056626393719587417",
+    "result": "313273654056626393719587417",
```
Comment (dbeal-eth) [inline packages/sources/synthetix-debt-pool/test/integration/__snapshots__/adapter.test.ts.snap:6]: these values are all going the opposite direction they should be--the debt is higher when there is synths in transit, not lower

**Review by dbeal-eth — DISMISSED** [review]
minor mistake on our end, looks great otherwise

**Review by dbeal-eth — APPROVED** [review]

**Comment by dbeal-eth** [comment]
its probably possible to query `getAddress` through the proxy itself because its a proxy, but the current implementation should work also

File: packages/sources/synthetix-debt-pool/src/utils.ts [hunk 3]
```diff
@@ -67,3 +77,41 @@ export const getContractAddress = async (
   const contractNameBytes32 = ethers.utils.formatBytes32String(contractName)
   return await addressResolver.getAddress(contractNameBytes32)
 }
+
+export const getSynthetixBridgeName = (networkName: string, jobRunID: string): string => {
+  if (networkName === SupportedChains.ETHEREUM || networkName === SupportedChains.KOVAN)
```
Comment (KuphJr) [inline packages/sources/synthetix-debt-pool/src/utils.ts:82]: Just a sanity check. Why would we return 'SynthetixBridgeToOptimism' if neworkName === SupportedChains.ETHEREUM and vis versa? Are these switched

Comment (justinkaseman) [reply packages/sources/synthetix-debt-pool/src/utils.ts:82]: Keyword is _to_ Optimism. So from Ethereum to Optimism.

**Review by KuphJr — APPROVED** [review]

**Pull request merged**